Source/items.cpp

```diff
@@ -3362,25 +3362,26 @@ void SpawnRewardItem(int itemid, Point position)
 	item._iSelFlag = 2;
 	item._iPostDraw = true;
 	item._iIdentified = true;
+	GenerateNewSeed(item);
 
-	if (gbIsMultiplayer) {
-		NetSendCmdPItem(false, CMD_DROPITEM, item.position, item);
+	if (sendmsg) {
+		NetSendCmdPItem(true, CMD_RESPAWNITEM, item.position, item);

```

Comment:
As an alternative to using `CMD_RESPAWNITEM` we could create a new command (`CMD_SPAWNITEM`) so that we have less special have handling in side the function. This also avoids reverting most of the changes. See https://github.com/diasurgical/devilutionX/pull/4198 for that option.